src/routes/Add.jsx

```diff
@@ -1,13 +1,129 @@
-import { Card, CardHeader } from '@nextui-org/react';
+import { useState } from 'react';
+import {
+  Card,
+  CardBody,
+  Input,
+  CardHeader,
+  Divider,
+  Select,
+  SelectItem,
+  Textarea,
+  Button
+} from '@nextui-org/react';
+import { serverTimestamp } from 'firebase/firestore';
+import LoadingCard from '../pages/Dictionary/components/LoadingCard';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import { unpackTermsQuery } from '../utils/unpackQuery';
+import getTermsQuery from '../utils/getTermsQuery';
+import getEntriesMutation from '../utils/getEntriesMutation';
 
 const Add = () => {
+  const [termSelectBox, setTermSelectBox] = useState('');
+  const [definitionTextArea, setDefinitionTextArea] = useState('');
+  const [exampleInput, setExampleInput] = useState('');
+  const [tagInput, setTagInput] = useState([]);
+  const navigate = useNavigate();
+
+  const [searchParams] = useSearchParams();
+  const searchEntries = Object.fromEntries(searchParams.entries());
+  const { term, ...search } = searchEntries;
+  const searchTerm = term !== undefined ? term.toLowerCase() : undefined;
+
+  const termsQuery = getTermsQuery(searchTerm);
+  const { status: termsStatus, data: terms } = unpackTermsQuery(termsQuery);
+
+  const mutation = getEntriesMutation();
+
+  const handleTermSelectionChange = (e) => {
+    setTermSelectBox(e.target.value);
+  };
+
+  const handleDefinitionTextAreaChange = (e) => {
+    setDefinitionTextArea(e.target.value);
+  };
+
+  const handleExampleInputChange = (e) => {
+    setExampleInput(e.target.value);
+  };
+
+  const handleTagInputChange = (e) => {
+    const tags = e.target.value.split(',').map((tag) => tag.trim());
+    setTagInput(tags);
+  };
+
+  const handleSubmit = () => {
+    mutation.mutate(
+      {
+        creationDate: serverTimestamp(),
+        definition: definitionTextArea,
+        example: exampleInput,
+        termid: termSelectBox,
+        likes: 0,
+        dislikes: 0,
+        tags: tagInput
+      },
+      {
+        onSuccess: () => {
+          navigate('/');
+        },
+        onError: (error) => {
+          console.error('Mutation error:', error);
+          // should add an alert...
+        }
+      }
+    );
+  };
+
   return (
     <section className="max-w-[55rem]">
-      <Card className="py-4">
-        <CardHeader className="flex-col">
-          <p className="font-bold">Add</p>
-        </CardHeader>
-      </Card>
+      {termsStatus === 'success' ? (
+        <Card className="py-4">
+          <CardHeader className="flex-col">
+            <p className="font-bold">New Definition</p>
+          </CardHeader>
+          <Divider />
+          <CardBody>
+            <div className="my-5">
+              <Select
```

Comment:
i would do `<Select disabled={termsStatus !== 'success'}/>` instead of returning LoadingCard so that a user can type other things while waiting for the loads.

Reply:
Nope. We need the terms. The user shouldn't be able to type anything if there isn't terms.